src/test/java/com/jcabi/w3c/DefaultHtmlValidatorITCase.java

```diff
@@ -45,21 +46,44 @@
 
     /**
      * DefaultHtmlValidator can validate HTML document.
+     *
      * @throws Exception If something goes wrong inside
      */
     @Test
     @RetryOnFailure(verbose = false)
     public void validatesHtmlDocument() throws Exception {
         MatcherAssert.assertThat(
-            ValidatorBuilder.HTML.validate(
-                StringUtils.join(
-                    "<!DOCTYPE html>",
-                    "<html><head><meta charset='UTF-8'/>",
-                    "<title>hey</title></head>",
-                    "<body></body></html>"
-                )
+                        ValidatorBuilder.HTML
+                                        .validate(this.validHtml()).errors(),
+                        Matchers.empty()
+        );
+    }
+
+    /**
+     * DefaultHtmlValidator can validate invalid HTML document.
+     *
+     * @throws Exception If something goes wrong inside
+     */
+    @Test
+    public void validatesInvalidHtmlDocument() throws Exception {
+        MatcherAssert.assertThat(
+                        ValidatorBuilder.HTML.validate(
+                                        "this is an invalid html"
             ).errors(),
-            Matchers.empty()
+            Matchers.not(Matchers.empty())
+        );
+    }
+
+    /**
+     * Creates a valid html.
+     * @return Valid html content
+     */
+    private String validHtml() {
```

Comment:
@jrdalpra first I prefer don't touch the code that does not refer to the ticket :)
You should create a new test to reproduce the issue and than fix it, if you have a time budget or create a puzzle otherwise.

yes if the code really unreadble, we can a helper methods.
but in this case the code was readable enough I think